src/main.rs

```diff
@@ -129,6 +131,66 @@ async fn start_servers(
     });
 }
 
+fn job_block_pruning(
+    schedule: &str,
+    cutoff_timestamp: u64,
+    block_store: BlockStore,
+) -> Result<Job, JobSchedulerError> {
+    Job::new_async(schedule, move |_uuid, _l| {
+        let block_store = block_store.clone(); // Get Arc for this job (can this clone be avoided?)
+        Box::pin(async move {
+            block_store.some_method().unwrap_or_else(|e| {
+                // TODO: handle error
+                eprintln!("Error running background job: {}", e);
+            });
+
+            let page_options = PageOptions {
+                page_size: Some(PAGE_SIZE_MAX),
+                ..PageOptions::default()
+            };
+
+            loop {
+                let mut total = 0u32;
+                let (count, done) = block_store
+                    .prune_until(&page_options, |header| header.timestamp >= cutoff_timestamp)
+                    .unwrap_or_else(|e| {
+                        // TODO: handle error
+                        eprintln!("Error pruning block store: {}", e);
+
+                        (0, true) // stop iterating if there's an error
+                    });
+                if done {
+                    break;
+                }
+
+                // TODO: better logging
+                total += count;
+                println!("Pruned {} blocks", total);
+            }
+        })
+    })
+}
+
+// TODO: handle shutdown_tx
+// TODO: does this need to be async?
+async fn schedule_background_jobs(app_config: &snapchain::cfg::Config, block_store: BlockStore) {
+    let sched = JobScheduler::new().await.unwrap();
+    if app_config.read_node {
+        if let Some(block_retention) = app_config.read_node_block_retention {
+            let cutoff_timestamp = (std::time::SystemTime::now() - block_retention)
+                .duration_since(std::time::UNIX_EPOCH)
```

Comment:
Header timestamps are in farcaster time (which uses a different epoch). You need to convert from farcaster time before you can compare.